SiVL19/get_iu_xray.py

```diff
@@ -49,47 +52,58 @@ def split_cases(reports_images, reports_text, keys, filename):
+images_auto_tags = {}
 reports_with_images = {}
 text_of_reports = {}
 
 for report in reports:
-
-	tree = ET.parse(os.path.join(reports_path, report))
-	root = tree.getroot()
-	img_ids = [] 
-	# find the images of the report
-	images = root.findall("parentImage")
-	# if there aren't any ignore the report
-	if len(images) == 0:
-		reports_with_no_image.append(report)
-	else:
-		sections = root.find("MedlineCitation").find("Article").find("Abstract").findall("AbstractText")
-		# find impression and findings sections
-		for section in sections:
-			if section.get("Label") == "FINDINGS":
-				findings = section.text
-			if section.get("Label") == "IMPRESSION":
-				impression = section.text
-
-		if impression is None and findings is None:
-			reports_with_empty_sections.append(report)
-		else:
-			if impression is None:
-				reports_with_no_impression.append(report)
-				caption = findings
-			elif findings is None:
-				reports_with_no_findings.append(report)
-				caption = impression
-			else:
-				caption = impression + " " + findings
-
-			for image in images:
-				images_captions[image.get("id") + ".png"] = caption
-				img_ids.append(image.get("id") + ".png")
-
-			reports_with_images[report] = img_ids
-			text_of_reports[report] = caption
-
+    tree = ET.parse(os.path.join(reports_path, report))
+    root = tree.getroot()
+    img_ids = []
+    # find the images of the report
+    images = root.findall("parentImage")
+    # if there aren't any ignore the report
+    if len(images) == 0:
+        reports_with_no_image.append(report)
+    else:
+        sections = root.find("MedlineCitation").find("Article").find("Abstract").findall("AbstractText")
+        # find impression and findings sections
+        for section in sections:
+            if section.get("Label") == "FINDINGS":
+                findings = section.text
+            if section.get("Label") == "IMPRESSION":
+                impression = section.text
+
+        if impression is None and findings is None:
+            reports_with_empty_sections.append(report)
+        else:
+            if impression is None:
+                reports_with_no_impression.append(report)
+                caption = findings
+            elif findings is None:
+                reports_with_no_findings.append(report)
+                caption = impression
+            else:
+                caption = impression + " " + findings
+
+            # get the MESH tags
+            tags = root.find("MeSH")
+            major_tags = []
+            auto_tags = []
+            if tags is not None:
+                major_tags = [t.text for t in tags.findall("major")]
+                auto_tags = [t.text for t in tags.findall("automatic")]
+
+            for image in images:
+                iid = image.get("id") + ".png"
+                images_captions[iid] = caption
+                img_ids.append(iid)
+                images_major_tags[iid] = major_tags
+                images_auto_tags[iid] = auto_tags
```

Comment:
Some reports do not contain automatic tags. Do we want to save the images of these reports?

Reply:
We want all images associated with a caption. Why should the caption existence matter?